### ISSUE TITLE
trigger opcodes

### ISSUE THREAD

**Issue by SeaPlayerPotato** [SoliaRdi/auto-fishing#10]
hi Soliardi , i play on sea and i trying to find opcodes but i cant find how to trigger opcodes for C_RQ_COMMIT_DECOMPOSITION_CONTRACT

 C_RQ_START_SOCIAL_PROGRESS_DECOMPOSITION

S_RESULT_DECOMPOSITION

can you help tell me how to trigger their opcodes

**Comment by Tera-Shiraneko** [comment]
I guess they can be triggered by dismantling items. First one if you open decomposition window second one if you dismatle them and the third one after dismantling.

**Comment by SeaPlayerPotato** [comment]
> I guess they can be triggered by dismantling items. First one if you open decomposition window second one if you dismatle them and the third one after dismantling.

i has been doing that but the opcode didnt coming up , that are why i confused , did i need more def file ? because i just using def file on this github ( i cant find like C_CAST_FISHING_ROD etc)

**Comment by Tera-Shiraneko** [comment]
Cant find eu opcodes too cause the packet logger isnt working for me. On ru it works to log them.

**Comment by SoliaRdi** [comment]
C_RQ_COMMIT_DECOMPOSITION_CONTRACT and C_CAST_FISHING_ROD and S_RESULT_DECOMPOSITION   is not needed. C_RQ_START_SOCIAL_PROGRESS_DECOMPOSITION is a simple packet, easy to log. [16:33:14:129] -> C_RQ_COMMIT_DECOMPOSITION_CONTRACT (id 23530)
{
    "contract": 824916
}
00000000: 0800 ea5b 5496 0c00                      ..j[T...

0800ea5b54960c00

**Comment by SeaPlayerPotato** [comment]
S_CHANGE_RELATION = 
S_USER_DEATH = 
C_CAST_FISHING_ROD = 
C_END_FISHING_MINIGAME = 37572
C_START_FISHING_MINIGAME = 26509
C_STOP_FISHING = 
S_CAST_FISHING_ROD = 
S_FISHING_CATCH = 63986
S_FISHING_STOP = 
S_FISHING_BITE = 63389
S_START_FISHING_MINIGAME = 61254
S_FISHING_CATCH_FAIL = 25674
C_RQ_ADD_ITEM_TO_DECOMPOSITION_CONTRACT = 30040
S_RP_ADD_ITEM_TO_DECOMPOSITION_CONTRACT = 32720
S_RP_START_SOCIAL_ON_PROGRESS_DECOMPOSITION = 
S_RP_COMMIT_DECOMPOSITION_CONTRACT = 
C_REQUEST_DECOMPOSITION = 60145
C_RQ_COMMIT_DECOMPOSITION_CONTRACT = 37887
C_RQ_START_SOCIAL_ON_PROGRESS_DECOMPOSITION =37129
S_RESULT_DECOMPOSITION =


only those opcodes i found till now, i gonna continue it later 😴

**Comment by SoliaRdi** [comment]
you need opcodes actually used in this mod) I just send all my opcodes mapped by hand, for some others mods

**Comment by SoliaRdi** [comment]
you need now S_RP_COMMIT_DECOMPOSITION_CONTRACT and S_RP_COMMIT_DECOMPOSITION_CONTRACT, thats all

**Comment by SoliaRdi** [comment]
used opcodes listed in manifest.json

**Comment by SeaPlayerPotato** [comment]
i cant find opcodes for 
S_RP_COMMIT_DECOMPOSITION_CONTRACT has been trying a lot times 

and
i dont get it after my caali proxy got updated (i also update my alex finder) i cant get any code from any other opcodes i has been ever had before

**Comment by Tera-Shiraneko** [comment]
Yeah thats because Owyn fucked his mod with bigInt change cause !== instead of equals() isnt working in that case. Also Caali rewrote tera-proxy-game that BigInt.prototype.toJSON = function() { return this.toString(); }; isnt working anymore cause it fucked up other modules if you got smth like that in your logger module it wont work too cause you have to replace this BigInt.prototype.toJSON = function() { return this.toString(); }; with an replacer function.

**Comment by SoliaRdi** [comment]
> Yeah thats because Owyn fucked his mod with bigInt change cause !== instead of equals() isnt working in that case. Also Caali rewrote tera-proxy-game that BigInt.prototype.toJSON = function() { return this.toString(); }; isnt working anymore cause it fucked up other modules if you got smth like that in your logger module it wont work too cause you have to replace this BigInt.prototype.toJSON = function() { return this.toString(); }; with an replacer function.
Well, i use this as tempolary solution in my logger
 function loopBigIntToString(obj){
		Object.keys(obj).forEach(key => {
				if (obj[key] && typeof obj[key] === 'object') loopNestedObj(obj[key]);
				else if(typeof obj[key]==="bigint") obj[key]=obj[key].toString();
		});
	}

**Comment by Tera-Shiraneko** [comment]
@SoliaRdi do you got discord or an email would be really appreciate it if I got someone who I can ask if I got problems with coding.

**Comment by SoliaRdi** [comment]
noname#0774

**Comment by Tera-Shiraneko** [comment]
Nice thank you.
